ppx/src/reactjs_jsx_ppx.ml

```diff
@@ -499,18 +500,30 @@ let jsxMapper =
             (Invalid_argument
                "JSX name can't be the result of function applications")
     in
-    let props =
-      Exp.apply ~loc (Exp.ident ~loc { loc; txt = propsIdent }) propsArg
+    let component = (nolabel, Exp.ident ~loc { txt = ident; loc })
+    and props =
+      ( nolabel,
+        Exp.apply ~loc (Exp.ident ~loc { loc; txt = propsIdent }) propsArg )
     in
-    let key_args =
-      match key with
-      | Some (label, key) ->
-          [ (label, mapper#expression ctxt key); (nolabel, unit) ]
-      | None -> []
-    in
-    Exp.apply ~loc ~attrs jsxExpr
-      ([ (nolabel, Exp.ident ~loc { txt = ident; loc }); (nolabel, props) ]
-      @ key_args)
+    match key with
+    | Some (label, key) ->
+        Exp.let_ ~loc Nonrecursive
```

Comment:
I would add a comment on why we need this `let` expression. Otherwise we (or at least I) will forget in just a few weeks 😅